tests/components/unifiled/test_config_flow.py

```diff
@@ -0,0 +1,97 @@
+"""Test the unifiled config flow."""
+from unittest.mock import patch
+
+from homeassistant import config_entries, setup
+from homeassistant.components.unifiled.const import DOMAIN
+from homeassistant.components.unifiled.config_flow import CannotConnect, InvalidAuth
+
+from tests.common import mock_coro
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.unifiled.config_flow.unifiled",
+        return_value=mock_coro({"title": "Test Title"}),
```

Comment:
This shouldn't return a mock coroutine. We can remove the setting of the return value as we're patching the unifiled class:
https://github.com/florisvdk/unifiled/blob/1eddc9b5b145bbd998cbc9e76ed06f7ae7e05d64/unifiled/__init__.py#L11

Reply:
Is there a good place to start learning how to write these tests. This is my first time working with them and i have a hard time understanding it. @MartinHjelmare

Reply:
I usually recommend this:

1. A blog on python testing eg:
https://realpython.com/python-testing/

2. The pytest docs:
https://docs.pytest.org/en/latest/

3. The `unittest.mock` library docs:
https://docs.python.org/3/library/unittest.mock.html

4. Look at existing tests, eg the tests for hue config_flow.

Reply:
I think i solved it.